p2p/discover/table.go

```diff
@@ -265,24 +293,23 @@ func (tab *Table) Lookup(targetID NodeID) []*Node {
 	return result.entries
 }
 
-func (tab *Table) requestRefresh() {
+func (tab *Table) refresh() {
+	done := make(chan struct{})
 	select {
-	case tab.refreshReq <- struct{}{}:
+	case tab.refreshReq <- done:
+		<-done
```

Comment:
This wait here makes the refresh blocking (duh :) ). My question really is if this could somehow stall other parts of the system while the refresh is running? Or to formulate it in a different way, why was the original `requestRefresh` async? Isn't there something we're regressing to?